src/parameters/add_parameters.jl

```diff
@@ -1,3 +1,17 @@
+"""
+Function to create a unique index of time series names for each device model. For example,
+if two parameters each reference the same time series name, this function will return a
+different value for each parameter entry
+"""
+function create_time_series_multiplier_index(
+    model,
+    ::Type{T},
+) where {T <: TimeSeriesParameter}
+    ts_names = get_time_series_names(model)
+    ts_name = ts_names[T]
```

Comment:
you can run a check here for the size of `ts_name` and only run line 12 is size > 1